objects/src/accounts/account_id.rs

```diff
@@ -140,14 +144,58 @@ impl AccountId {
         ];
         let mut current_digest = Self::compute_digest(current_seed, code_root, storage_root);
 
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Generating new account seed [pow={}, current_digest={}, current_seed={}]",
+                digest_pow(current_digest),
+                digest_hex(current_digest),
+                word_hex(current_seed),
+            );
+        }
+
+        #[cfg(feature = "log")]
+        let mut count = 0;
+        #[cfg(feature = "log")]
+        let mut best: (u32, Digest, Word) =
+            (digest_pow(current_digest), current_digest, current_seed);
+
         // loop until we have a seed that satisfies the specified account type.
         loop {
+            #[cfg(feature = "log")]
+            {
+                count += 1;
+                let current = (digest_pow(current_digest), current_digest, current_seed);
+                if best.0 <= current.0 {
+                    best = current;
+                }
+
+                if count % 500_000 == 0 {
+                    log::debug!(
+                        "Account seed loop [count={}, best_pow={}, best_digest={}, best_seed={}]",
+                        count,
+                        best.0,
+                        digest_hex(best.1),
+                        word_hex(best.2),
+                    );
+                }
+            }
```

Comment:
I would probably move this into a separate function (e.g., `update_pow_progress()`) to keep the body of this function cleaner. `update_pow_progress` could take mutable references to `count` and `best`, and we could probably also combine them into a single tuples. For example, the tuple could be `(count, digest, seed)`.

Reply:
moved logging and state out of this function